src-input/duk_bi_reflect.c

```diff
@@ -0,0 +1,110 @@
+/*
+ *  'Reflect' built-in (ES7 Section 26.1)
+ *  http://www.ecma-international.org/ecma-262/7.0/#sec-reflect-object
+ */
+
+#include "duk_internal.h"
+
+#if defined(DUK_USE_REFLECT_BUILTIN)
+DUK_INTERNAL duk_ret_t duk_bi_reflect_object_delete_property(duk_context *ctx) {
+	duk_hthread *thr;
+	duk_hobject *hobj;
+	duk_hstring *key;
+	duk_bool_t ret;
+
+	DUK_ASSERT_TOP(ctx, 2);
+	thr = (duk_hthread *) ctx;
+	DUK_ASSERT(thr != NULL);
+
+	/* [ target key ] */
+
+	hobj = duk_require_hobject(ctx, 0);
+	key = duk_to_hstring(ctx, 1);
+
+	ret = duk_hobject_delprop_raw(thr, hobj, key, 0 /*flags*/);
+	duk_push_boolean(ctx, ret);
+	return 1;
+}
+
+DUK_INTERNAL duk_ret_t duk_bi_reflect_object_get(duk_context *ctx) {
+	duk_hthread *thr;
+	duk_tval *tv_obj;
+	duk_tval *tv_key;
+	duk_idx_t nargs;
+
+	thr = (duk_hthread *) ctx;
+	DUK_ASSERT(thr != NULL);
+
+	nargs = duk_get_top(ctx);
+	if (nargs < 2) {
+		DUK_DCERROR_TYPE_INVALID_ARGS((duk_hthread *) ctx);
+	}
+
+	/* [ target key receiver? ...? ] */
+
+	(void) duk_require_hobject(ctx, 0);
+	tv_obj = duk_require_tval(ctx, 0);
+	(void) duk_to_string(ctx, 1);
```

Comment:
Agree, the code also reads much cleaner that way (better separation of concerns).